src/NHSD.BuyingCatalogue.Ordering.Api/Services/ContactDetailsService.cs

```diff
@@ -29,9 +32,13 @@ public Contact AddOrUpdatePrimaryContact(Contact existingContact, PrimaryContact
             if (existingContact is null)
                 return newOrUpdatedContact.ToDomain();
 
-            context.Entry(existingContact).CurrentValues.SetValues(newOrUpdatedContact);
-            existingContact.Email = newOrUpdatedContact.EmailAddress;
-            existingContact.Phone = newOrUpdatedContact.TelephoneNumber;
+            if (newOrUpdatedContact is not null)
+            {
+                existingContact.FirstName = newOrUpdatedContact.FirstName;
+                existingContact.LastName = newOrUpdatedContact.LastName;
+                existingContact.Email = newOrUpdatedContact.EmailAddress;
+                existingContact.Phone = newOrUpdatedContact.TelephoneNumber;
+            }
```

Comment:
As above, let's invert the `if` here (to reduce the amount of nested code).

Reply:
Inverted `If` statement to check `Null`